### PR TITLE
Cloud Modules STS Support Redux

### PR DESCRIPTION
**Pull request by semifocused**
Add support for AWS Security Token Service (temporary credentials) to all AWS cloud modules.

Rationalize sources and ordering of security credentials for consistency across modules.

Update documentation to use shared doc fragment and consistent usage information across modules.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "lib/ansible/modules/core"]
 	path = lib/ansible/modules/core
-	url = https://github.com/ansible/ansible-modules-core.git
+	url = https://github.com/47lining/ansible-modules-core.git
```
Comment (resmo) [inline .gitmodules]: Please review your pull request once again.

File: lib/ansible/modules/core [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 8f6ae92cf88beda287c6c11d8b4127239c3168e0
+Subproject commit e8a56653c113c492092a5177a3574b24c2b3e5d4
```
Comment (resmo) [inline lib/ansible/modules/core]: Please review your pull request once again.

Comment (semifocused) [inline lib/ansible/modules/core]: Gotcha, I'll get a branch together without those commits - we need them, but you don't :)

**Comment by semifocused** [comment]
I've created https://github.com/ansible/ansible/pull/9894 instead, easier to get just the desired changes onto a new branch.  Closing this PR.

**Pull request closed without merging**